app/api_topologies_test.go

```diff
@@ -50,6 +52,95 @@ func TestAPITopology(t *testing.T) {
 	}
 }
 
+func TestAPITopologyAddsContainerLabelFilters(t *testing.T) {
+	ts := topologyServer()
+	defer ts.Close()
+
+	body := getRawJSON(t, ts, "/api/topology")
+
+	var topologies []app.APITopologyDesc
+	decoder := codec.NewDecoderBytes(body, &codec.JsonHandle{})
+	if err := decoder.Decode(&topologies); err != nil {
+		t.Fatalf("JSON parse error: %s", err)
+	}
+	equals(t, 4, len(topologies))
+
+	topologyRegistry := &app.Registry{
+		Items: map[string]app.APITopologyDesc{},
+	}
+
+	for _, topology := range topologies {
+		topologyRegistry.Add(topology)
+	}
+
+	var containerOpts []app.APITopologyOption
+	containerOpts = append(containerOpts, app.MakeFilterOption("all", "All", nil, false))
+	containerOpts = append(containerOpts, app.MakeFilterOption("system", "System Containers", render.IsSystem, false))
+	containerOpts = append(containerOpts, app.MakeFilterOption("notsystem", "Application Containers", render.IsApplication, false))
+	containerFilters := []app.APITopologyOptionGroup{
+		app.MakeAPITopologyOptionGroup("system", "application", containerOpts),
+		app.MakeAPITopologyOptionGroup("stopped", "running",
+			[]app.APITopologyOption{
+				app.MakeFilterOption("stopped", "Stopped containers", render.IsStopped, false),
+				app.MakeFilterOption("running", "Running containers", render.IsRunning, false),
+				app.MakeFilterOption("both", "Both", nil, false),
+			},
+		),
+		app.MakeAPITopologyOptionGroup("pseudo", "hide",
+			[]app.APITopologyOption{
+				app.MakeFilterOption("show", "Show Uncontained", nil, false),
+				app.MakeFilterOption("hide", "Hide Uncontained", render.IsNotPseudo, true),
+			},
+		),
+	}
+
+	unconnectedFilter := []app.APITopologyOptionGroup{
+		app.MakeAPITopologyOptionGroup("unconnected", "hide",
+			[]app.APITopologyOption{
+				// Show the user why there are filtered nodes in this view.
+				// Don't give them the option to show those nodes.
+				app.MakeFilterOption("hide", "Unconnected nodes hidden", nil, false),
+			},
+		),
+	}
+
+	topologyRegistry.Add(
+		app.MakeAPITopologyDesc(
+			"processes",
+			render.FilterUnconnected(render.ProcessWithContainerNameRenderer),
+			"Processes",
+			1,
+			unconnectedFilter,
+			true,
+		),
+		app.MakeAPITopologyDescWithParent(
+			"processes-by-name",
+			"processes",
+			render.FilterUnconnected(render.ProcessNameRenderer),
+			"by name",
+			0,
+			unconnectedFilter,
+			true,
+		),
+		app.MakeAPITopologyDesc(
+			"containers",
+			render.ContainerWithImageNameRenderer,
+			"Containers",
+			2,
+			containerFilters,
+			false,
+		),
+	)
+
+	equals(t, 3, topologyRegistry.CountContainerFilters())
+
+	v := app.MakeFilterOption(fmt.Sprintf("cmdlinefilter%d", 0), "title", render.HasLabel("role=system"), false)
+	topologyRegistry.AddContainerFilters(v)
+
+	equals(t, 4, topologyRegistry.CountContainerFilters())
```

Comment:
This test is not testing the extra functionality you added. You should be testing that your parsing works as expected and that filtering works (i.e. rendering only includes containers with the criteria of the label filter when applied).